executing/executing.py

```diff
@@ -522,10 +522,7 @@ def add_qualname(self, node, name=None):
 
     def visit_FunctionDef(self, node, name=None):
         # type: (ast.AST, Optional[str]) -> None
-        if sys.version_info[0] == 3:
-            assert isinstance(node, ast.FunctionDef) or isinstance(node, ast.AsyncFunctionDef) or isinstance(node, ast.Lambda), node
-        else:
-            assert isinstance(node, ast.FunctionDef) or isinstance(node, ast.Lambda), node
+        assert isinstance(node, (ast.FunctionDef, ast.Lambda)) or sys.version_info[0] == 3 and isinstance(node, ast.AsyncFunctionDef), node
```

Comment:
Need to revert this, as doing it the `if` way makes the `node` types locked down. Otherwise, we get `executing/executing.py:532: error: "AST" has no attribute "body"` because the type in the `else` of the Lambda if later on is "any sort of AST", because mypy isn't quite smart enough